modules/hal_nxp/CMakeLists.txt

```diff
@@ -1,12 +1,17 @@
 #
 # Copyright (c) 2021 Linaro, Limited
+# Copyright 2024 NXP
 #
 # SPDX-License-Identifier: Apache-2.0
 #
 
 if(CONFIG_HAS_MCUX OR CONFIG_HAS_IMX_HAL OR CONFIG_HAS_NXP_S32_HAL)
   add_subdirectory(${ZEPHYR_CURRENT_MODULE_DIR} hal_nxp)
   add_subdirectory_ifdef(CONFIG_USB_DEVICE_DRIVER usb)
+  add_subdirectory_ifdef(CONFIG_UHC_DRIVER usb)
+  if (CONFIG_USB_UHC_NXP_MCUX)
```

Comment:
Can we use a more generic Kconfig, like `CONFIG_NXP_FSL_OSA`, to control inclusion of this file?

Reply:
Thanks Daniel, I have updated the codes, please have a check.